### PR TITLE
Conditionally load the Monaco CDN.

### PR DESCRIPTION
**Pull request by savindi7**
### Related Issues
- 

### Related PRs
- 

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/index.tsx [hunk 1]
```diff
@@ -36,6 +37,32 @@ import Theme from "./theme";
 // Set the runtime config in the context.
 ContextUtils.setRuntimeConfig(Config.getDeploymentConfig());
 
+// Function to check the status of the Monaco CDN.
+// If the CDN is not available, the default CDN will be used.
+async function checkCDNStatus() {
```
Comment (brionmario) [inline apps/console/src/index.tsx:42]: Lets use arrow functions and link a Git Issue with a `TODO`.
```suggestion
async function checkCDNStatus() {
```

File: apps/console/src/index.tsx [hunk 2]
```diff
@@ -36,6 +37,32 @@ import Theme from "./theme";
...
+            });
+        } else {
+            loader.config({
+                paths: {
+                    vs: "https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/0.26.1/min/vs"
+                }
+            });
+        }
+    } catch (error) {
+        //Use default CDN.
```
Comment (brionmario) [inline apps/console/src/index.tsx:60]: What's the fallback ?

File: apps/console/src/index.tsx [hunk 3]
```diff
@@ -36,6 +37,32 @@ import Theme from "./theme";
 // Set the runtime config in the context.
 ContextUtils.setRuntimeConfig(Config.getDeploymentConfig());
 
+// Function to check the status of the Monaco CDN.
+// If the CDN is not available, the default CDN will be used.
+async function checkCDNStatus() {
+    try {
+        const response: Response = await fetch("https://cdn.jsdelivr.net/npm/monaco-editor@0.46.0/min/vs/loader.js");
```
Comment (pavinduLakshan) [inline apps/console/src/index.tsx:44]: Can't we do something like keeping a local copy of monaco editor in the `public` folder and referencing to it from here?

File: apps/console/src/index.tsx (hunk 1 above)
Comment (savindi7) [reply apps/console/src/index.tsx]: fixed with https://github.com/wso2/identity-apps/pull/6255/commits/c7d66702d38416b4e7c78e3a5ce8c98dc3c30d9a

**Comment by savindi7** [comment]
Ran the bundle analyzer

before
Screenshot 2024-05-27 at 18 06 37

After 
Screenshot 2024-05-27 at 18 06 37

File: apps/console/src/index.tsx [hunk 4]
```diff
@@ -36,6 +37,37 @@ import Theme from "./theme";
...
+            });
+        } else {
+            loader.config({
+                paths: {
+                    vs: "https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/0.36.1/min/vs"
+                }
+            });
+        }
+    } catch (error) {
+        // Use default CDN.
```
Comment (pavinduLakshan) [inline apps/console/src/index.tsx:65]: When the internet is not reachable, we can configure to load a locally available loader script to be loaded in the catch block.

To do this, we first need to add "loader.js" to console/public folder.

```suggestion
        // Use default CDN.        
        const response: Response = await fetch(contextPathGlobal + "loader.js");

        if (response.ok) {
            loader.config({
                paths: {
                    vs: contextPathGlobal + "loader.js"
                }
            })
        }
```

@brionmario what are your thoughts on this?

Comment (savindi7) [reply apps/console/src/index.tsx:65]: loader just a script used to download Monaco resources, we will still need internet for this AFIK...

File: apps/console/src/index.tsx (hunk 3 above)
Comment (savindi7) [reply apps/console/src/index.tsx]: Refer comment below

File: apps/console/src/index.tsx (hunk 4 above)
Comment (pavinduLakshan) [reply apps/console/src/index.tsx:65]: If the script is packed with console bundle, the app will not need to access internet to get the script.

Comment (brionmario) [reply apps/console/src/index.tsx:65]: @pavinduLakshan So your suggestion is to maintain a `minified` monaco bundle as a static file? I hope `loader.js` is just a generic term used for the actual monaco minified bundle?

Is there a one single such minified bundle available to download?

Comment (pavinduLakshan) [reply apps/console/src/index.tsx:65]: > Is there a one single such minified bundle available to download?

We can download it from https://cdn.jsdelivr.net/npm/monaco-editor@0.36.1/min/vs or extract from the monaco-editor npm package, can't we?

Comment (savindi7) [reply apps/console/src/index.tsx:65]: https://cdn.jsdelivr.net/npm/monaco-editor@0.36.1/min/vs/loader.js this is the path the loader which is a script used to download the Monaco resources. We will still need internet access to get the Monaco resources AFIK. Could you clarify this suggestion a bit more? @pavinduLakshan

Comment (pavinduLakshan) [reply apps/console/src/index.tsx:65]: If the loader.js script is just for downloading stuff from internet, we wouldn't be able to follow my suggested approach. Sorry for the noise.

File: apps/console/src/index.tsx (hunk 2 above)
Comment (DonOmalVindula) [reply apps/console/src/index.tsx]: Let's log an error using `console.error` so that we can be aware if it fails

**Review by DonOmalVindula — APPROVED** [review]

**Pull request merged**